jvm-packages/xgboost4j-spark/src/main/scala/ml/dmlc/xgboost4j/scala/spark/params/LearningTaskParams.scala

```diff
@@ -33,6 +33,18 @@ private[spark] trait LearningTaskParams extends Params {
 
   final def getObjective: String = $(objective)
 
+  /**
+   * The learning objective type of the specified custom objective and eval.
+   * Corresponding type will be assigned if custom objective is defined
+   * options: regression, classification. default: null
+   */
+  final val obj_type = new Param[String](this, "obj_type", "objective type used for " +
+    s"training, options: {${LearningTaskParams.supportedObjectiveType.mkString(",")}",
+    (value: String) => LearningTaskParams.supportedObjectiveType.contains(value))
+
+  final def getObjectiveType: String = $(obj_type)
```

Comment:
it still failed, you need to set a default value for obj_type but interpret it properly when the user didn't set customized objective func